### PR TITLE
API Sample Refactor

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR refactors the API sample documentation.

## Review Changes

💻 [Preview](https://deploy-preview-1215--docs-spectrocloud.netlify.app/api/samples)

🎫 [Jira Ticket](https://spectrocloud.atlassian.net/jira/software/c/projects/DOC/boards/62?modal=detail&selectedIssue=DOC-729)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1215#" title="Author self-approved">karl-cardenas-coding</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/api/2-samples.mdx [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+## Prerequisites
+
+* You must have a Spectro Cloud account. If you do not have an account, you can create one at [https://console.spectrocloud.com](https://console.spectrocloud.com).
+
+
+* An Authentication header with a token value or an API Key with an ApiKey value. 
+Learn more about authentication credentials by reviewing the [authentication methods](https://docs.spectrocloud.com/user-management/user-authentication) resource.
+
+
+* You must have a Spectro Cloud cluster. You can create a cluster from the Spectro Cloud console.
```
Comment (ritawatson) [inline content/api/2-samples.mdx:36]:
```suggestion
* You must have a Palette cluster. You can create a cluster in Palette.
```

File: content/api/2-samples.mdx [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+
+* The respective language runtime and package manager installed on your machine. 
+
+
+
+* [jq](https://stedolan.github.io/jq/download/) - A command-line JSON processor.
+
+## Configure Scope and Authentication
+
+In the following examples, the following variables are used to interact with the Palette API. The values for these variables are specific to your environment.
```
Comment (ritawatson) [inline content/api/2-samples.mdx:47]:
```suggestion
Variables shown in the following examples are used to interact with the Palette API. The values for these variables are specific to your environment.
```

File: content/api/2-samples.mdx [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+else:
+  # Print the error message if the request failed
+  print("Request failed with status code:", response.status_code)
+```
+
+</Tabs.TabPane>
+
+<Tabs.TabPane tab="Rust" key="node-status-rust">
+
+```
```
Comment (ritawatson) [inline content/api/2-samples.mdx:220]: Is rust needed here?
```rust

File: content/api/2-samples.mdx [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+  print(f"UID: {uid}")
+else:
+  # Print the error message if the request failed
+  print("Request failed with status code:", response.status_code)
+```
+</Tabs.TabPane>
+
+<Tabs.TabPane tab="Rust" key="config-identifier-rust">
+
+```
```
Comment (ritawatson) [inline content/api/2-samples.mdx:444]: Is ``rust`` after the three tick marks?

File: content/api/2-samples.mdx [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+
+</Tabs.TabPane>
+
+
+</Tabs>
+
+##  Retrieve the Cluster Cloud Config Identifier
+
+The cloud config identifier is a unique identifier for the cloud config that is used to provision the cluster. 
+The values are found in the `spec.cloudConfigRef` field of the cluster object. You can use the following code snippet to retrieve the cluster cloud config identifier `uid` and `kind`.
```
Comment (ritawatson) [inline content/api/2-samples.mdx:268]:
```suggestion
The values are found in the `spec.cloudConfigRef` field of the cluster object. You can use the following code snippet to retrieve the `uid` and `kind` for the cluster cloud config identifier. 
```

**Review by ritawatson — APPROVED** [review]
LGTM with a couple comments.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1215#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1215#pullrequestreview-1362046501" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by prathabk — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1215#" title="Author self-approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1215#pullrequestreview-1366222391" title="Approved">prathabk</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1215#pullrequestreview-1362046501" title="Approved">ritawatson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/api/2-samples.mdx [hunk 6]
```diff
@@ -0,0 +1,1855 @@
...
+
+**Endpoint**: `https://api.spectrocloud.com/v1/spectroclusters/{provider}`
+
+Set the provider as an environment variable. 
+
+```shell
+export PROVIDER="Your Provider"
+```
+
+The payload for creating a host cluster can be lengthy due to the various configurations available for a host cluster. To improve readbility, the payload is condensed in the following examples and lack required values.
```
Comment (karl-cardenas-coding) [inline content/api/2-samples.mdx:92]:
```suggestion
The payload for creating a host cluster can be lengthy due to the various configurations available for a host cluster. The payload is condensed in the following examples to improve readability and lack all the required values. 
```

**Pull request merged**